branchff

```diff
@@ -116,6 +116,16 @@ logrun mkdir -p $WORKDIR
 gitlib::sync_repo $K8S_GITHUB_SSH $TREE_ROOT || common::exit 1 "Exiting..."
 logrun cd $TREE_ROOT
 
+# Is the master branch in a reasonable state to fast forward over the branch?
+GIT_DESCRIBE=$(git describe)
+if [[ ! $GIT_DESCRIBE =~ ^v${RELEASE_BRANCH/release-/} ]]; then
```

Comment:
ah, I think I figured out the necessary git-fu:
To find the common ancestor of two branches (i.e. where we branched or merged from):
```
$ git merge-base master release-1.7
fc582d4b21d06e5ebfe85d4737b42fa9f3868021
```

You can then maybe use this with `git tag --contains` to figure out what you want:
```
$ git tag --contains fc582d4b21d06e5ebfe85d4737b42fa9f3868021
$ git checkout master
$ git tag v18donotpush
$ git tag --contains fc582d4b21d06e5ebfe85d4737b42fa9f3868021
v18donotpush
$ git checkout release-1.7
$ git tag v17donotpush
$ git tag --contains fc582d4b21d06e5ebfe85d4737b42fa9f3868021
v17donotpush
v18donotpush
```